iOS Client/Middleware/MBMiddleware.swift

```diff
@@ -24,4 +24,22 @@ struct MiddlewareFactory {
     }
     
     // Next Middleware . . .
+    static let saveDevotionMiddleware: Middleware<Any> = { dispatch, getState in
```

Comment:
Yo this middle ware syntax kind of confuses me, but if I'm understanding correctly it will try to save the devotion and if it does, then it will continue to next middleware, if not then it will not pass on?

Reply:
that's right. If there's an error, it swallows it and doesn't even go to the reducers . . . which kind of haunts me a bit. B/c, if a user clicks a table view cell, and then we can't mark the devotion as read b/c an error occurs here, then they don't go anywhere, and they don't get an error. An error would be very unlikely . . . maybe worth dispatching a new action in the event of an error that triggers an alert view or something . . .

Reply:
I altered this to be best-effort. Saving as a pure side-effect. So, even if it fails, I print an error but let the UDF continue so the user doesn't perceive any issues.

They just might notice that the red dot doesn't go away